### PR TITLE
chore: improve CircleCI config

### PR DESCRIPTION
**Pull request by jackmu95**
### Describe the PR

This PR improves the CircleCI config and brings it more in line with the one from Nuxt.js.

### PR checklist

**What kind of change does this PR introduce?** (check at least one)

- [ ] Bugfix
- [ ] Feature
- [x] Enhancement
- [ ] ARIA accessibility
- [ ] Documentation update
- [ ] Other (please describe)

**Does this PR introduce a breaking change?** (check one)

- [x] No
- [ ] Yes (please describe)

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch, **not** the `master` branch
- [ ] When resolving a specific issue, it's referenced in the PR's title (i.e. `[...] (fixes #xxx[,#xxx])`, where "xxx" is the issue number)
- [x] It should address only one issue or feature. If adding multiple features or fixing a bug and adding a new feature, break them into separate PRs if at all possible.
- [x] The title should follow the [**Conventional Commits**](https://www.conventionalcommits.org/) naming convention (i.e. `fix(alert): not alerting during SSR render`, `docs(badge): update pill examples, fix typos`, `chore: fix typo in README`, etc). **This is very important, as the `CHANGELOG` is generated from these messages.**

**If new features/enhancement/fixes are added or changed:**

- [ ] Includes documentation updates (including updating the component's `package.json` for slot and event changes)
- [ ] New/updated tests are included and passing (if required)
- [ ] Existing test suites are passing
- [ ] The changes have not impacted the functionality of other components or directives
- [ ] ARIA Accessibility has been taken into consideration (Does it affect screen reader users or keyboard only users? Clickable items should be in the tab index, etc.)

**If adding a new feature, or changing the functionality of an existing feature, the PR's
description above includes:**

- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -1,89 +1,114 @@
...
-            - setup
-          filters:
-            branches:
-              only:
-                - master
+      - lint:  { requires: [setup] }
+      - audit: { requires: [setup] }
+      - build: { requires: [setup] }
+      - lint:  { requires: [build] }
+      - test:  { requires: [build] }
```
Comment (tmorehouse) [inline .circleci/config.yml]: should this be `test-unit` instead of `test`?

Comment (jackmu95) [reply .circleci/config.yml]: Renamed the job back to `test`.

**Comment by tmorehouse** [comment]
Looks like it can't find coverage for the tests...

Hopefully this will also fix the issue with merging #3559

**Comment by jackmu95** [comment]
@tmorehouse Yeah, I hope to fix #3559 with this PR.

**Comment by tmorehouse** [comment]
codecov is not telling the truth, so we can ignore it...

Still need to figure out the deploy-preview hook issue... (it is a webhook, and not part of CI config)

**Comment by jackmu95** [comment]
Running `yarn docs-gen` locally gives me the same error.

**Comment by tmorehouse** [comment]
Strange...

Maybe it needs setup and build?

**Comment by tmorehouse** [comment]
`dev` branch `docs-gen` works fine...

maybe something in yarn.lock has changed?

**Comment by jackmu95** [comment]
@tmorehouse Yeah, regenerated it to fix some security vulnerabilities I found by running `yarn audit`.

**Comment by tmorehouse** [comment]
Might be a Nuxt compatibility issue with some newer versions?

**Pull request merged**